avr/cores/megacommand/MCL/GridTrack.cpp

```diff
@@ -36,12 +28,22 @@ bool GridTrack::load_from_grid(uint8_t column, uint16_t row, bool data) {
   case EXT_TRACK_TYPE:
     ::new(this) ExtTrack;
     break;
-  default:
-    // unrecognized track type
+  }
+  return true;
+}
+
+bool GridTrack::load_from_grid(uint8_t column, uint16_t row, bool data) {
+
+  bool ret;
+
+  ret = proj.read_grid(this, sizeof(GridTrack), column, row);
+  if (!ret) {
+    DEBUG_PRINTLN("read failed");
     return false;
   }
 
-  if (!data) return true;
+ if (!data) return true;
+ if (!init_track_type(active)) return false;
```

Comment:
init track first, or virtual functions will fail on (!data)

Reply:
how so?

Reply:
grid_track.load_from_grid(..., false);
grid_track.get_track_size(..) // vtable not initialized...